src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -81,21 +83,50 @@ public static FunctionPointerMethodSymbol CreateMethodFromSource(FunctionPointer
                 callingConvention,
                 refKind,
                 returnType,
+                refReadonlyModifiers,
                 syntax,
                 typeBinder,
                 diagnostics,
                 suppressUseSiteDiagnostics);
         }
 
+        public static FunctionPointerMethodSymbol CreateMethodFromMetadata(CallingConvention callingConvention, ImmutableArray<ParamInfo<TypeSymbol>> retAndParamTypes)
+        {
+            Debug.Assert(retAndParamTypes.Length > 0);
+
+            ParamInfo<TypeSymbol> retInfo = retAndParamTypes[0];
```

Comment:
>ParamInfo<TypeSymbol> retInfo = retAndParamTypes[0] [](start = 12, length = 51)

Why can't we do all this return type related work within the constructor?

Reply:
We could. But given that Chuck just asked me to move what _was_ in the constructor into a helper method, I'm going to leave this where it is.

---
In reply to: [374405675](https://github.com/dotnet/roslyn/pull/41340#discussion_r374405675) [](ancestors = 374405675)

Reply:
> We could. But given that Chuck just asked me to move what was in the constructor into a helper method, I'm going to leave this where it is.

I am not sure how one is related to the other. I think, from the API design perspective, it would be better if constructor were taking all data in raw form, or all data in "preprocessed" form.


---
In reply to: [374952548](https://github.com/dotnet/roslyn/pull/41340#discussion_r374952548) [](ancestors = 374952548,374405675)